Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStore.mm

```diff
@@ -110,7 +110,6 @@
         return;
 
     m_parameters = parameters;
-    clearBackingStore();
```

Comment:
This is a bit iffy, I can't remember this exactly.
This might be needed in some other cases, like webgl?

Reply:
The general pattern is that we either call `prepareForDisplay` (which drops any buffer handles, and ensures a following `createContextAndPaintContents` to paint contents instead), or we just set a new buffer handle.

Dropping existing buffer handles when we notify of a size/format/opaque-ness change etc shouldn't be necessary. It does mean that paths that use buffer handles need to re-set their handle more often than otherwise needed though, which is why I dropped it here.

This is all complicated (imo) due to the 'backing store' also handling the 'external buffer' path. I think we should pull all the delegate/buffer-handle code out into a separate object, and make the layer own one or the other.